### PR TITLE
Add `opt_out` column to `expt_assignments` table

### PR DESCRIPTION
**Pull request by karlhigley**
This also renames the table from `expt_allocations` to `expt_assignments`. That felt more urgent now that we're going to have additional tables (like `datasets` and `account_aliases`) that will connect to related tables like `accounts`.

### CODE REVIEW

File: poprox-db/migrations/versions/2024_07_26_0945-fc8d79481ee4_merge_head_revisions.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+"""merge head revisions
```
Comment (kluver) [inline poprox-db/migrations/versions/2024_07_26_0945-fc8d79481ee4_merge_head_revisions.py:1]: Yeah, I manually made the pre-commit fix revisions in the past, but I think there is an argument that we should maybe just exclude these?

Comment (kluver) [reply poprox-db/migrations/versions/2024_07_26_0945-fc8d79481ee4_merge_head_revisions.py:1]: I have code that ignores the `poprox-db/migrations/versions` folder when doing code style checks. I'm holding back committing pending your thoughts on the topic.

Comment (karlhigley) [reply poprox-db/migrations/versions/2024_07_26_0945-fc8d79481ee4_merge_head_revisions.py:1]: No strong opinion either way, but I went ahead and applied the formatting/fixes from the linters

Comment (kluver) [reply poprox-db/migrations/versions/2024_07_26_0945-fc8d79481ee4_merge_head_revisions.py:1]: The default template has code style errors in it (unused imports I think) so I think it might be worth it. If you have no strong opinions I think I'm going to go ahead and do it.

**Comment by kluver** [comment]
I was hoping to downstream some changes from this PR (so I didn't have to deal with the alembic merge) -- I don't think this passes the database tests.

**Comment by kluver** [comment]
> I was hoping to downstream some changes from this PR (so I didn't have to deal with the alembic merge) -- I don't think this passes the database tests.

```
    if isinstance(revision.down_revision, tuple):
        return  # don't test merge revisions
```
Adding this to the staircase test "fixes" it -- it does prevent testing merge revisions, but I think we can plan to always have empty merge revisions
